activesupport/lib/active_support/error_reporter.rb

```diff
@@ -28,14 +28,15 @@ class ErrorReporter
     DEFAULT_SOURCE = "application"
     DEFAULT_RESCUE = [StandardError].freeze
 
-    attr_accessor :logger, :debug_mode
+    attr_accessor :logger, :debug_mode, :context_middlewares
```

Comment:
I think we'd rather expose a `add_middleware` method or something like that, and that should be enough as a public API.

Reply:
Done. I moved the documentation from the inner class to `add_middleware` since that's the main point of interaction.